lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -200,6 +277,85 @@ void ClangImporter::recordModuleDependencies(
   }
 }
 
+void ClangImporter::recordBridgingHeaderOptions(
+    ModuleDependencyInfo &MDI,
+    const clang::tooling::dependencies::TranslationUnitDeps &deps) {
+  auto &ctx = Impl.SwiftContext;
+
+  std::vector<std::string> swiftArgs;
+  auto addClangArg = [&](Twine arg) {
+    swiftArgs.push_back("-Xcc");
+    swiftArgs.push_back(arg.str());
+  };
+
+  // We are using Swift frontend mode.
+  swiftArgs.push_back("-frontend");
+
+  // Swift frontend action: -emit-pcm
+  swiftArgs.push_back("-emit-pch");
```

Comment:
We need to consider carefully what is added here versus what is added in the driver's `generatePCHJob`, which also includes `addCommonFrontendOptions`, which may have additional context on top of what we know here.

Reply:
I can post the patch for swift driver too. I haven't got time to write extensive tests for that.